torch/_dynamo/variables/torch.py

```diff
@@ -1222,7 +1230,11 @@ def speculate_branch(branch):
             example_value = deepcopy_to_fake_tensor(example_res, tx.fake_mode)
 
             p_args = (lowered_node,) + p_args
-        elif self.value.__name__ in ("wrap", "wrap_activation_checkpoint"):
+        elif self.value.__name__ in (
+            "wrap",
+            "wrap_activation_checkpoint",
+            "tag_activation_checkpoint",
```

Comment:
curious where do we generate these `wrap` or `tag` ac calls inside dynamo? is the dynamo detecting `torch.utils.checkpoint` and convert this call to a highorderop?

Reply:
Yes, we do this inside Dynamo. This PR is actually doing exactly that. It detects torch.utils.checkpoint and converts it into higherorderop - https://github.com/pytorch/pytorch/blob/main/torch/_dynamo/utils.py#L1681